lightstep/resource_metric_condition_test.go

```diff
@@ -784,3 +785,71 @@ func TestValidateGroupBy(t *testing.T) {
 		}
 	}
 }
+
+func Test_buildLatencyPercentiles(t *testing.T) {
+	type args struct {
+		lats    []interface{}
+		display string
+	}
+	tests := []struct {
+		name string
+		args args
+		want []float64
+	}{
+		{
+			name: "Full List Provided; Expect Full List",
+			args: args{
+				lats:    []interface{}{float64(50), float64(95), float64(99), 99.9},
+				display: "line",
+			},
+			want: []float64{50, 95, 99, 99.9},
+		},
+		{
+			name: "Partial List Provided; Expect Partial List",
+			args: args{
+				lats:    []interface{}{float64(50), float64(95)},
+				display: "line",
+			},
+			want: []float64{50, 95},
+		},
+		{
+			name: "No List Provided; Expect Full List",
+			args: args{
+				lats:    []interface{}{},
+				display: "line",
+			},
+			want: []float64{50, 95, 99, 99.9},
+		},
+		{
+			name: "Heatmap: Full List Provided; Expect No List",
+			args: args{
+				lats:    []interface{}{float64(50), float64(95), float64(99), 99.9},
+				display: "heatmap",
+			},
+			want: []float64{},
+		},
+		{
+			name: "Heatmap: Partial List Provided; Expect No List",
+			args: args{
+				lats:    []interface{}{float64(50), float64(95)},
+				display: "heatmap",
+			},
+			want: []float64{},
+		},
+		{
+			name: "Heatmap: No List Provided; Expect No List",
+			args: args{
+				lats:    []interface{}{},
+				display: "heatmap",
+			},
+			want: []float64{},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := buildLatencyPercentiles(tt.args.lats, tt.args.display); !reflect.DeepEqual(got, tt.want) {
```

Comment:
`require.Equal` calls DeepEqual under the hood. I think it's more idiomatic to use `require.Equal` though in our unit tests.

Reply:
Interesting. This boiler-plate is what IntelliJ creates when using `Generate test for function`. You're thinking like this?
`for _, tt := range tests {
    t.Run(tt.name, func(t *testing.T) {
	got := buildLatencyPercentiles(tt.args.lats, tt.args.display)
	require.Equal(t, got, tt.want, "buildLatencyPercentiles() = %v, want %v", got, tt.want)
    })
}`

Reply:
`require.Equal` will output the diff for you, so I think this could be simplified further to:
```
for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			got := buildLatencyPercentiles(tt.args.lats, tt.args.display)
                         require.Equal(tt, got, tt.want)
                 }
}
```